airtime_mvc/application/common/AutoPlaylistManager.php

```diff
@@ -39,40 +39,45 @@ public static function buildAutoPlaylist() {
             $full = false;
             $repeatuntilfull = $playlistrepeat->getAutoPlaylistRepeat();
             $tempPercentScheduled = 0;
+            $si = new Application_Model_ShowInstance($autoplaylist->getDbId());
+            // the intro playlist should be added exactly once
+            if ($introplaylistid != null) {
+                //Logging::info('adding intro');
+                $si->addPlaylistToShowStart($introplaylistid, false);
+            }
             while(!$full) {
-                $si = new Application_Model_ShowInstance($autoplaylist->getDbId());
                 // we do not want to try to schedule an empty playlist
                 if ($playlistid != null) {
                     $si->addPlaylistToShow($playlistid, false);
                 }
                 $ps = $si->getPercentScheduled();
-                if ($introplaylistid != null) {
-                    //Logging::info('adding intro');
-                    $si->addPlaylistToShowStart($introplaylistid, false);
-                }
-                if ($outroplaylistid != null) {
-                    //Logging::info('adding outro');
-                    $si->addPlaylistToShow($outroplaylistid, false);
-                    //Logging::info("The total percent scheduled is % $ps");
-                }
                 if ($ps > 100) {
```

Comment:
Can this check not check the current duration plus the duration of the outro is less that the show duration? That way it won't run over if repeat until full is selected

Reply:
There is no easy way to determine what the effect of adding a playlist to a show will be until it happens. This is because the playlist could have a random smartblock. 
We could "disable" the outro playlist if repeat until full is enabled. There is a basic conflict between repeat until full and the outro playlist that I didn't realize until the software scheduled hours of filler tracks instead of repeating the same random selection.

It's not ideal but I think moving the intro and outro out of the loop is the quickest and most logical fix. The intro and outro playlist are so hidden in the settings that it is unlikely that they are widely in use and they aren't well documented in terms of how they are supposed to work.

I think it makes sense that the intro is added once to the start of the show and the outro once to the end vs. appending them multiple times which would have happened with the repeat until full w/o a fill time remaining in either of them.

I guess the only other potential conflict introduced by this is if there is a "limit to time remaining" smartblock in the intro it could supercede the content, whereas I had coded that intro to be inserted "after" the content playlist was repeated and now it has been moved to happen first.

We don't currently have the tools coded to "check" the contents of the playlist to see whether it contains a "limit to time remaining smartblock".

The whole notion of playlists and smartblocks is pretty confusing from an UI perspective and I'd like to revise it but I think this fix is probably the right way to prevent the scenario I highlighted in the bug.

Reply:
Ah right, I had forgotten about that.LGTM then